src/python/tests/seq_region/test_collection.py

```diff
@@ -0,0 +1,366 @@
+
+    Args:
+        input_seq: Sequence dict with usable values (`codon_table`, `location`).
+        code_map: A custom map location -> codon table number.
+        expected_codon_table: Expected codon table number.
+    """
+    collection = SeqCollection()
+    seq_name = "foobar"
+    collection.seqs = {seq_name: input_seq}
+    collection.add_translation_table(code_map)
+    assert collection.seqs[seq_name].get("codon_table") == expected_codon_table
+
+
+@pytest.mark.parametrize(
+    "input_seq, taxon_id, response_data, expected_codon_table, expected",
+    [
+        param(
+            {"location": "mitochondrial_chromosome"},
+            99,
+            '{"mitochondrialGeneticCode": 4}',
+            4,
+            no_raise(),
+            id="Add codon table",
+        ),
+        param(
+            {"location": "mitochondrial_chromosome", "codon_table": 2},
+            99,
+            '{"mitochondrialGeneticCode": 4}',
+            2,
+            no_raise(),
+            id="Existing codon table",
+        ),
+        param({"location": "mitochondrial_chromosome"}, 99, "{}", None, no_raise(), id="No code found"),
+        param({}, 99, '{"mitochondrialGeneticCode": 4}', None, no_raise(), id="Missing location"),
+        param({"location": "mitochondrial_chromosome"}, 0, "{}", None, no_raise(), id="No taxon_id to use"),
+        param(
+            {"location": "mitochondrial_chromosome"},
+            99,
+            "<html></html>",
+            None,
+            raises(ValueError),
+            id="Not a JSON response, HTML",
+        ),
+    ],
+)
+@patch("ensembl.io.genomio.seq_region.collection.requests.get")
+def test_add_mitochondrial_codon_table(
+    mock_requests_get: Mock,
+    mock_response: Callable,
+    input_seq: dict[str, str],
+    taxon_id: int,
+    response_data: str,
+    expected_codon_table: int | None,
+    expected: ContextManager,
+) -> None:
+    """Test `SeqCollection.add_mitochondrial_codon_table()`.
+
+    Args:
+        input_seq: Sequence dict with usable values (`codon_table`, `location`).
+        taxon_id: Taxon ID passed to the request.
+        response_data: Return data from the request.
+        expected_codon_table: Expected codon table after update.
+        expected: Context manager to catch expected exceptions.
+    """
+
+    mock_requests_get.return_value = mock_response(response_data)
+    collection = SeqCollection()
+    seq_name = "foobar"
+    collection.seqs = {seq_name: input_seq}
+    with expected:
+        collection.add_mitochondrial_codon_table(taxon_id)
+        assert collection.seqs[seq_name].get("codon_table") == expected_codon_table
+
+
+@patch("ensembl.io.genomio.seq_region.collection.requests.get")
+def test_add_mitochondrial_codon_table_mock(mock_requests_get: Mock) -> None:
+    """Test `SeqCollection.add_mitochondrial_codon_table()` with the mock parameter."""
+    mock_requests_get.return_value = MockResponse("{}")  # Mock requests just in case
```

Comment:
```suggestion
def test_add_mitochondrial_codon_table_mock(mock_requests_get: Mock, mock_response: MockResponse) -> None:
    """Test `SeqCollection.add_mitochondrial_codon_table()` with the mock parameter."""
    mock_requests_get.return_value = mock_response("{}")  # Mock requests just in case
```
_Note:_ not sure about the type hint in this case, or if something needs to be imported.